### PR TITLE
Initial adaptation from Jupyter to matplotlib

### PR DESCRIPTION
**Pull request by mdboom**
This is a first pass at adapting the Jupyter content for matplotlib.

### CODE REVIEW

File: governance.md [hunk 1]
```diff
@@ -68,26 +65,27 @@ The foundations of Project governance are:
...
+consistent contributions have been recognized by their receiving
+“commit rights” to the Project repositories. In general all Project
+decisions are made through consensus among the Core Developers with
+input from the Community.
+
+While this approach has served us well, as the Project grows and faces
+more legal and financial decisions and interacts with other
+institutions, we see a need for a more formal governance model. Moving
+forward The Project leadership will consist of a Steering Council with
+the power to appoint final decisions to an individual. We view this
```
Comment (efiring) [inline governance.md]: Something is wrong here--I'm not sure what you intended to say.

File: governance.md [hunk 2]
```diff
@@ -225,26 +223,27 @@ most of the time, and explicit BDFL intervention is sought only if the
...
     NumFOCUS and the direction of the Project as determined by the full
     Council.
 -   This Subcommittee shall NOT make decisions about the direction, scope or
     technical direction of the Project.
--   This Subcommittee will have 5 members, 4 of whom will be current Council
-    Members and 1 of whom will be external to the Steering Council. No more
-    than 2 Subcommitee Members can report to one person through employment or
-    contracting work (including the reportee, i.e. the reportee + 1 is the
-    max). This avoids effective majorities resting on one person.
+-   This Subcommittee will have 4 members, at least 2 of which are
```
Comment (efiring) [inline governance.md]: which -> whom

Comment (efiring) [inline governance.md]: Should we expand to 5?  Having an odd number matters only in majority-vote situations.

File: governance.md (hunk 1 above)
Comment (dopplershift) [inline governance.md]: Yeah, I'm not sure how "with the power to appoint final decisions to an individual" is better than "a BDFL", as was in the original text from jupyter.

**Comment by dopplershift** [comment]
There's a phrase "The BDFL can appointing his/her successor" (came over from Jupyter, so not in the diff) that should also be cleaned up.

Otherwise, I have no problem with this. I think the model has served Jupyter well to this point.

File: governance.md (hunk 1 above)
Comment (tacaswell) [inline governance.md]: Sorry, we started trying to remove the BDFL and then went back on it, but forget to revert this change.

File: governance.md (hunk 2 above)
Comment (tacaswell) [inline governance.md]: I agree we should expand this to 5, but 4 is what matches the current state of the world.  I suggest that we expand it to 5 at some point later using the power this documents gives us to mutate this document.

**Comment by mdboom** [comment]
I've just pushed some changes that I hope address the comments here.

Additionally, on a second read I decided to elaborate on the history section a bit.  It draws the linkage from John through Mike to Thomas, so it isn't quite so jarring when Thomas is first mentioned in the BDFL section.

File: people.md [hunk 3]
```diff
@@ -2,52 +2,23 @@
...
-- [UC Berkeley](http://www.berkeley.edu/) (Fernando Perez)
-- [Rackspace](http://www.rackspace.com/) (Kyle Kelley)
-
-### Tier 2
-
-- [Google](https://www.google.com/)
+- Thomas Caswell
+- Michael Droettboom
+- Phil Elson
+- Eric Firing
```
Comment (mdboom) [inline people.md]: Subcommittee members must have contiguous last initials, apparently.

File: governance.md [hunk 4]
```diff
@@ -4,46 +4,43 @@ The official version of this document, along with a list of
...
+The Matplotlib Project (The Project) is an open source software
+project affiliated with the 501c3 NumFocus Foundation. The goal of The
+Project is to develop open source software and deploy open and public
+websites and services for data visualization. The Software developed
+by The Project is released under the BSD (or similar permissive) open
+source license, developed openly and hosted in public GitHub
+repositories under the
+[Matplotlib GitHub organization](https://github.com/matplotlib). Examples
+of Project Software include the matplotlib library for data
+visualization and associated extensions and dependencies.  The
```
Comment (WeatherGod) [inline governance.md]: "...and associated extensions and dependencies"? what does this mean, specifically? This is really vague.

File: governance.md [hunk 5]
```diff
@@ -4,46 +4,43 @@ The official version of this document, along with a list of
...
+Development Team (MDT)" in the project license.  Anyone can be a
+Contributor. Contributors can be affiliated with any legal entity or
+none. Contributors participate in the project by submitting, reviewing
+and discussing GitHub Pull Requests and Issues and participating in
+open and public Project discussions on GitHub, Google+, Hackpad,
+Gitter chat rooms and mailing lists. The foundation of Project
+participation is openness and transparency.
 
-Here is a list of the current Contributors to the main IPython repository:
+Here is a list of the current Contributors to the main Matplotlib repository:
```
Comment (WeatherGod) [inline governance.md]: Should we note that this is just a subset of what we call "Contributors"?

File: governance.md [hunk 6]
```diff
@@ -68,26 +65,29 @@ The foundations of Project governance are:
...
+of Contributors, called Core Developers, whose active and consistent
+contributions have been recognized by their receiving “commit rights”
+to the Project repositories. In general all Project decisions are made
+through consensus among the Core Developers with input from the
+Community.
+
+While this approach has served us well, as the Project grows and faces
+more legal and financial decisions and interacts with other
+institutions, we see a need for a more formal governance model. Moving
+forward The Project leadership will consist of a Benevolent Dictator
```
Comment (WeatherGod) [inline governance.md]: missing comma after "forward".

File: governance.md [hunk 7]
```diff
@@ -293,42 +294,27 @@ building proprietary products and services that leverage Project Software and
...
-To acknowledge institutional contributions, there are two level of Institutional
-Partners, with associated benefits:
+To acknowledge institutional contributions, Institutional Partners
+receive the following benefits:
 
-**Tier 1** = an institution with at least one Institutional Council Member
-
--   Acknowledged on the Jupyter/IPython websites, in talks and T-shirts.
--   Ability to acknowledge their own funding sources on the Jupyter/IPython
+-   Acknowledged on the Matplotlib websites, in talks and T-shirts.
```
Comment (WeatherGod) [inline governance.md]: Do we need to hedge these statements a bit? For example, a few months ago, I gave a talk where I used matplotlib's development process as an example of how OSS works to people who were looking to possibly adopt a community-driven development model. With this, would I have been required to include these institutions in my talk?

Another example... lightning talks? Tutorials? Also, talks by whom? Would there be a difference if I gave a tutorial (as a Core Developer) versus a Community Member?

File: governance.md [hunk 8]
```diff
@@ -293,42 +294,27 @@ building proprietary products and services that leverage Project Software and
...
 
 Changing the Governance Documents
 =================================
 
 Changes to the governance documents are submitted via a GitHub pull
 request to The Project's governance documents GitHub repository at
-[https://github.com/jupyter/governance](https://github.com/jupyter/governance).
+[https://github.com/matplotlib/governance](https://github.com/matplotlib/governance).
 The pull request is then refined in response to public comment and
 review, with the goal being consensus in the community.  After this
```
Comment (WeatherGod) [inline governance.md]: Probably should add a line to the effect of "For substantial changes, a notice will be posted to the matplotlib-users mailing list"?

File: governance.md (hunk 4 above)
Comment (tacaswell) [inline governance.md]: I think the vaugeness is intentional.  I would lump cycler into this category.

Should talk to @kthyng and @astrofrog about lumping cmocean and pytest-mpl into this category.

This is something that I think asking the jupyter folks about how they are / plan to deal with the large number of extensions they have.

Comment (astrofrog) [inline governance.md]: If pytest-mpl ends up being used as part of the matplotlib test suite, then I think it definitely belongs in this category.

File: governance.md (hunk 7 above)
Comment (tacaswell) [inline governance.md]: I would interpret this as being talks _about_ matplotlib the library would fall under this, talks about process would not, and lightning talks are a gray area.  Any talk about mpl should acknowledge the (many) other contributors.   This should not be any more onerous than acknowledging your grants in scientific talks.

I suspect that we would end up with a stock image/slide with the proper set of logos.

File: governance.md (hunk 8 above)
Comment (tacaswell) [inline governance.md]: I am disinclined to change this wording too much.  If we add the phrase 'substantial' in, then we would have to sort out 'substantial' means, etc.

If we end up where this document is in force and the Steering Council is trying to change it in secret, it is time for a fork.

File: governance.md (hunk 5 above)
Comment (mdboom) [inline governance.md]: That's a good point.  This doesn't include people who contribute in non-code-commit ways.

File: governance.md (hunk 7 above)
Comment (mdboom) [inline governance.md]: What about "talks substantively about matplotlib given by matplotlib Contributors"?

We can't really stop someone from giving a talk or tutorial about matplotlib at saying that it was created by magical unicorns.  I think this is more a statement about talks given by core contributors.  I like @tacaswell's idea of making the proper ackowledgement as easy as possible.

File: governance.md (hunk 4 above)
Comment (kthyng) [inline governance.md]: What does this category change mean?

Comment (pelson) [inline governance.md]: I'm inclined to agree with @WeatherGod here - we could be specific and refer to repositories under the matplotlib GH org. Are there repos for which this doesn't apply (e.g. Basemap)?

File: governance.md (hunk 5 above)
Comment (tacaswell) [inline governance.md]: I have had some discussion with a jupyter core member (@Carreau) about the limitations of 'contributor' in this document.

Comment (Carreau) [inline governance.md]: Here is a short recap of some discussion _I_ had with various people (regardless of whether they are part of Jupyter  or not), this may not represent the feeling of the Jupyter team in general, and has been reworded through my own French brain, so might contain inaccurate terms, but I hope to convey the right meaning:

- Like to recognize contribution of non-code / non-github more.
- Like to get non-coder in the governance
- Like to get more diversity in the governance
- Nomination to the steering by steering member feel a lot like we get only people we agree with.
- People that spend a lot of time on the project can get a narrow view of its usage which is non-representative.
- Conflation between "reward for having participated for a year" and "Suited to lead the project". Should there be a separate reward for long-sustained contribution ?

I'll try to catch up with the rest of the discussion.

File: governance.md [hunk 9]
```diff
@@ -4,46 +4,44 @@ The official version of this document, along with a list of
...
-([http://jupyter.org](http://jupyter.org) and
-[http://ipython.org](http://ipython.org)), nbviewer
-([https://nbviewer.ipython.org](https://nbviewer.ipython.org)) and the
-Jupyter coLaboratory
-([https://colaboratory.jupyter.org](https://colaboratory.jupyter.org)).
+The Matplotlib Project (The Project) is an open source software
+project affiliated with the 501c3 NumFocus Foundation. The goal of The
+Project is to develop open source software and deploy open and public
+websites and services for data visualization. The Software developed
+by The Project is released under the BSD (or similar permissive) open
```
Comment (QuLogic) [inline governance.md]: Assuming this covers everything under the Matplotlib GitHub organization, basemap bundles GEOS (LGPL) and GMT data (GPL), and natgrid's license is "Restricted" (whatever that means), which I think don't strictly fall under "permissive".

Comment (tacaswell) [inline governance.md]: The only weird thing about natgrid is `src/nncrunch.c`. The main source is BSD-2 clause and @jswhit 's code is MIT (I think?).

Is there a way build basemap without bundling those dependencies?  

thoughts @WeatherGod ?

Comment (WeatherGod) [inline governance.md]: with respect to basemap and GEOS, the source tree merely vendors a version of the library, but doesn't require that it be used, IIRC. As for the GMT data, I have always been highly skeptical of the GPL claim for the data. The software (which is not included) is GPL, and that is fine, but I have had a hard time believing anyone can even apply GPL to a dataset.

As for unbundling the data from basemap, it is technically possible (conda-forge supposedly does it), but no one has submitted a patch upstream to do it there as well. However, basemap is pretty much useless without that data. It is theoretically possible to use a different dataset (Cartopy uses another dataset), but converters would need to be written to generate them into the internal format used by basemap (which is pretty much undocumented).

File: governance.md [hunk 10]
```diff
@@ -4,46 +4,44 @@ The official version of this document, along with a list of
...
-Project consist of public websites and web-services that are hosted
-under the jupyter.org or ipython.org domains. The Services Examples of
-Project Services include the Jupyter and IPython websites
-([http://jupyter.org](http://jupyter.org) and
-[http://ipython.org](http://ipython.org)), nbviewer
-([https://nbviewer.ipython.org](https://nbviewer.ipython.org)) and the
-Jupyter coLaboratory
-([https://colaboratory.jupyter.org](https://colaboratory.jupyter.org)).
+The Matplotlib Project (The Project) is an open source software
+project affiliated with the 501c3 NumFocus Foundation. The goal of The
```
Comment (OceanWolf) [inline governance.md]: Living outside of the US, I had no idea what 501c3 meant, I took it as part of the name of the organisation..

Why not write  "affiliated with the non-profit organisation, NumFocus Foundation, covered by section 501(c)(3) of the United States Internal Revenue Code".

Comment (OceanWolf) [inline governance.md]: Oh, and you should probably link to the website here... http://www.numfocus.org/

File: governance.md (hunk 4 above)
Comment (OceanWolf) [inline governance.md]: I don't think it matters either way, we talk about "examples", which means we can allow for vagueness here I guess... oh, an grammar mistake...

and **its** associated extensions

File: governance.md [hunk 11]
```diff
@@ -4,46 +4,44 @@ The official version of this document, along with a list of
...
+and discussing GitHub Pull Requests and Issues and participating in
+open and public Project discussions on GitHub, Google+, Hackpad,
+Gitter chat rooms and mailing lists. The foundation of Project
+participation is openness and transparency.
 
-Here is a list of the current Contributors to the main IPython repository:
+Here is a list of the current Contributors to the main Matplotlib repository:
 
-[https://github.com/ipython/ipython/graphs/contributors](https://github.com/ipython/ipython/graphs/contributors)
+[https://github.com/matplotlib/matplotlib/graphs/contributors](https://github.com/matplotlib/matplotlib/graphs/contributors)
```
Comment (OceanWolf) [inline governance.md]: Confused, as this link only gives 100 people (with no pagination controls), but https://github.com/matplotlib/matplotlib states "480 contributors" at time of writing...

File: governance.md [hunk 12]
```diff
@@ -68,26 +66,29 @@ The foundations of Project governance are:
...
-need for a more formal governance model. Moving forward The Project leadership
-will consist of a BDFL and Steering Council. We view this governance model as
-the formalization of what we are already doing, rather than a change in
-direction.
+Project leadership was provided by the original author, John
+D. Hunter.  Shortly before his passing in 2012, leadership was
+transferred to Michael Droettboom, who later invited Thomas Caswell as
+a co-lead.  Additional leadership has also been provided by a subset
+of Contributors, called Core Developers, whose active and consistent
+contributions have been recognized by their receiving “commit rights”
```
Comment (OceanWolf) [inline governance.md]: this does not make sense... from my reading it seems like you mean to say that an individual receiving "commit rights" indicates acknowledgement of their active and consistent contributions.

As the text stands it should read...

>[their] contributions have been recognized by **someone**, which has led to them receiving commit rights to...

but that still sounds clunky...

>Core Developers provide additional leadership and have "commit rights" to the Project repositories; Core Developers comprise of a subset of Contributors whose active and consistent contributions have been recognised by **someone**.

I still feel a little confused here by the language... by ``Core Leadership`` do you refer to those listed on https://github.com/orgs/matplotlib/teams as belonging to the ``Core`` group, or to us ``Developers`` as well.  From the description it sounds like you refer to all of us in the ``Developers`` team, but then you don't list @efiring and @jenshnielsen as part of the Project Leadership.  We should ensure consistency between the names we use here and the names of the teams in the project.

File: governance.md (hunk 4 above)
Comment (mdboom) [inline governance.md]: Fixed @OceanWolf's grammar suggestion.

File: governance.md (hunk 9 above)
Comment (mdboom) [inline governance.md]: Should we just explicitly exclude basemap as falling under different rules (wrt licensing) so we can move forward with everything else?

File: governance.md (hunk 12 above)
Comment (mdboom) [inline governance.md]: In this text, "Core Developers" refers to everyone in the "Developers" github team.  I think I should just change the text here.  The meaning of the "Core" team on github will probably need to be refined.  (Right now, it's sort of "Numfocus subcommittee" + regular meeting attendees, but that's not a definition I think we necessarily want to keep).

I think to move forward, I'll just correct the text here to match the github team name, and we can refine the definition of Core later if we want/need to.

**Comment by mdboom** [comment]
All: I've made the final "simple" corrections and merged this.

I think it was most important to begin to have something in effect than waiting for perfect.  Specifically, I think @Carreau's discussion about "broader" community, and @OceanWolf's points about more specifically defining certain groups are good points, and those discussions can continue in other issues and PRs.

Thanks everyone for helping with this, and congrats, @tacaswell.
